eng/scripts/Generate-Patch.ps1

```diff
@@ -151,24 +153,59 @@ function ResetSourcesToReleaseTag($ArtifactName, $ServiceDirectoryName, $Release
   if(Test-Path $TestResourcesFilePath) {
     $cmdOutput = git restore --source $ReleaseTag -W -S $TestResourcesFilePath
   }
+  
+  if($SkipCodeQuality) {
+    ## We need to update the POM file to have the skip coverage flags.
+    $pomFilePath = Join-Path $ArtifactDirPath "pom.xml"
+    if(!(Test-Path $pomFilePath)) {
+      LogError "Could not find the pom file to update the skip code quality tags."
+      exit 1
+    }
 
-  if(Test-Path $CheckStyleSuppressionFilePath) {
-    $cmdOutput = git restore --source $ReleaseTag -W -S $CheckStyleSuppressionFilePath
-  }
+    $pomFileContent = New-Object System.Xml.XmlDocument
+    $pomFileContent.PreserveWhitespace = $true
 
-  if(Test-Path $CheckStyleFilePath) {
-    $cmdOutput = git restore --source $ReleaseTag -W -S $CheckStyleFilePath
-  }
+    $pomFileContent.Load($pomFilePath)
+    $projectNode = $pomFileContent.project
+    if(!$projectNode) {
+      LogError "Failed to log the elements"
+      exit 1
+    }
 
-  if(Test-Path $SpotBugsFilePath) {
-    $cmdOutput = git restore --source $ReleaseTag -W -S $SpotBugsFilePath
-  }
+    $propertiesNode = $projectNode.properties
+	$namespaces = $projectNode.NamespaceURI
+    $pomFileContent.PreserveWhitespace = $false
+    if(!$propertiesNode) {
+      $propertiesNode = $pomFileContent.CreateElement("properties", $namespaces)
+      $dependenciesNode = $projectNode.GetElementsByTagName("dependencies")
+      $orderedPropertiesTag = $pomFileContent.DocumentElement.InsertBefore($propertiesNode, $dependenciesNode[0])
+    }
+
+    $checkstyleNodeElement = $propertiesNode.AppendChild($pomFileContent.CreateElement("checkstyle.skip", $namespaces))
+    $checkstyleNodeValue = $checkstyleNodeElement.AppendChild($pomFileContent.CreateTextNode("true"))
+    $spotBugsNodeElement = $propertiesNode.AppendChild($pomFileContent.CreateElement("spotbugs.skip", $namespaces))
+    $spotBugsNodeValue = $spotBugsNodeElement.AppendChild($pomFileContent.CreateTextNode("true"))
+    $pomFileContent.Save($pomFilePath)
```

Comment:
Do we really have to modify the pom file? Can we accomplish the same thing by passing in a command line parameter?

Reply:
I really wanted to avoid this as well but chose this instead because of the following
1. We will have to create separate pipelines just for the patch releases in order to pass extra arguments.
2. It will be easily reproable on a local machine and would not require pipeline magic.

Is there a better way to achieve these?